### ISSUE TITLE
com.android.internal.telephony.PhoneFactory

### ISSUE THREAD

**Issue by xjpmauricio** [anggrayudi/android-hidden-api#37]
Is it possible to access com.android.internal.telephony.PhoneFactory ?

**Comment by Trumeet** [comment]
This is an `internal` API, this project only provides `Framework` hidden API now, such as `android.app.StatusBarManager` and etc.
Such as those APIs below are **NOT** included in `Framework APIs`:
`com.android.internal.*`
`com.android.server.*` (in service jar)

`com.android.telephony.PhoneFactory` is included in `telephony` jar. So it not included in this project now. You can try compile AOSP yourself and find those jars in out folder.

**Comment by Trumeet** [comment]
You can download services jars in #36 .

**Comment by anggrayudi** [comment]
Yes — `com.android.internal.telephony.PhoneFactory` is included when you build your own jar. I confirmed the class `com/android/internal/telephony/PhoneFactory.class` is present in a CLI-built `android.jar`.

This repo now ships a CLI, **`hiddenjar`**, that builds a custom `android.jar` (with hidden/internal APIs) directly from a running emulator or device — for any API level:

```bash
./cli/hiddenjar build --api <level>                 # macOS/Linux
# Windows (native PowerShell, no Git Bash): powershell -ExecutionPolicy Bypass -File cli\hiddenjar.ps1 build --api <level>
# or any OS via Gradle:
./gradlew buildHiddenJar -Papi=<level>
```

It merges the whole `$BOOTCLASSPATH` (framework.jar **plus** the mainline/APEX modules) and verifies the result compiles a hidden API before finishing. See the README section **“Build Your Own Custom `android.jar` (CLI)”**.

_Note: this unblocks **compilation**. Runtime access to non-SDK APIs is still subject to the Android 9+ (API 28+) hidden-API blocklist for non-system apps._

Closing — reopen if a CLI-built jar doesn't solve it for you.
